examples/models/object_detection/yolox/pascal_voc/train.py

```diff
@@ -0,0 +1,456 @@
+        image, gt_boxes, gt_classes = resize_fn(image, gt_boxes, gt_classes)
+        gt_classes = tf.expand_dims(gt_classes, axis=-1)
+        bounding_boxes = tf.concat([gt_boxes, gt_classes], axis=-1)
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes, images=image, source="yxyx", target=bounding_box_format
+        )
+        return {"images": image, "bounding_boxes": bounding_boxes}
+
+    return apply
+
+
+train_ds = train_ds.map(
+    proc_train_fn("xywh", image_size), num_parallel_calls=tf.data.AUTOTUNE
+)
+train_ds = train_ds.apply(
+    tf.data.experimental.dense_to_ragged_batch(BATCH_SIZE, drop_remainder=True)
+)
+eval_ds = eval_ds.map(
+    proc_train_fn(bounding_box_format="xywh", img_size=image_size),
+    num_parallel_calls=tf.data.AUTOTUNE,
+)
+eval_ds = eval_ds.apply(
+    tf.data.experimental.dense_to_ragged_batch(BATCH_SIZE, drop_remainder=True)
+)
+
+"""
+Looks like everything is structured as expected.  Now we can move on to constructing our
+data augmentation pipeline.
+"""
+train_ds = train_ds.prefetch(2)
+train_ds = train_ds.shuffle(BATCH_SIZE)
+eval_ds = eval_ds.prefetch(2)
+
+
+def unpackage_dict(inputs):
+    return inputs["images"], inputs["bounding_boxes"]
+
+
+train_ds = train_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(unpackage_dict, num_parallel_calls=tf.data.AUTOTUNE)
+
+"""
+Our data pipeline is now complete.  We can now move on to model creation and training.
+"""
+
+"""
+## Model creation
+
+We'll use the KerasCV API to construct a YoloX_tiny model.  In this tutorial we use
+a pretrained ResNet50 backbone using weights.  In order to perform fine-tuning, we
+freeze the backbone before training.  When `include_rescaling=True` is set, inputs to
+the model are expected to be in the range `[0, 255]`.
+"""
+
+model = keras_cv.models.YoloX_tiny(
+    classes=20,
+    bounding_box_format="xywh",
+    backbone="cspdarknet",
+    include_rescaling=True,
+    evaluate_train_time_metrics=False,
+)
+
+metrics = [
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        class_ids=range(21),
+        bounding_box_format="xywh",
+        name="Mean Average Precision",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(21),
+        bounding_box_format="xywh",
+        max_detections=100,
+        name="Recall",
+    ),
+]
+optimizer = tf.optimizers.SGD(global_clipnorm=10.0)
+model.compile(
+    classification_loss="binary_crossentropy",
+    objectness_loss="binary_crossentropy",
+    box_loss="iou",
+    optimizer=optimizer,
+    metrics=metrics,
+)
+
+callbacks = [
+    keras.callbacks.TensorBoard(log_dir="logs"),
+    keras.callbacks.ReduceLROnPlateau(patience=5),
+    keras.callbacks.EarlyStopping(patience=10),
+    keras.callbacks.ModelCheckpoint(CHECKPOINT_PATH, save_weights_only=True),
+]
+
+history = model.fit(
+    train_ds,
+    validation_data=eval_ds.take(10),
```

Comment:
same comment as above, let's start with custom training loop so you can evaluate the entire eval_ds after each epoch, or after several epochs